src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/CodeActions/CSharp/TypeAccessibilityCodeActionProvider.cs

```diff
@@ -152,6 +171,59 @@ internal class TypeAccessibilityCodeActionProvider : CSharpCodeActionProvider
             return true;
         }
 
+        private static bool TryProcessCodeActionVS(
+            RazorCodeActionContext context,
+            CodeAction codeAction,
+            Diagnostic diagnostic,
+            string associatedValue,
+            out ICollection<CodeAction> typeAccessibilityCodeActions)
+        {
+            CodeAction processedCodeAction = null;
+
+            // When there's only one FQN suggestion, code action title is of the form:
+            // `System.Net.Dns`
+            if (!codeAction.Title.Any(c => char.IsWhiteSpace(c)) &&
+                codeAction.Title.EndsWith(associatedValue, StringComparison.OrdinalIgnoreCase))
+            {
+                var fqn = codeAction.Title;
+                processedCodeAction = CreateFQNCodeAction(context, diagnostic, codeAction, fqn);
+            }
+            // When there are multiple FQN suggestions, the code action title is of the form:
+            // `Fully qualify 'Dns'`
+            else if (codeAction.Title.Equals($"Fully qualify '{associatedValue}'", StringComparison.OrdinalIgnoreCase))
+            {
+                // Not currently supported as we need O# CodeAction to support the CodeAction.Children field.
+                // processedCodeAction = codeAction.WrapResolvableCSharpCodeAction(context, LanguageServerConstants.CodeActions.FullyQualifyType);
+
+                typeAccessibilityCodeActions = Array.Empty<CodeAction>();
+                return false;
+            }
+            // For add using suggestions, the code action title is of the form:
+            // `using System.Net;`
+            else if (AddUsingVSCodeAction.Match(codeAction.Title).Success)
+            {
+                var @namespace = AddUsingsCodeActionProviderFactory.ExtractNamespaceFromVSCSharpAddUsing(codeAction.Title);
```

Comment:
```suggestion
            else if (AddUsingVSCodeAction.Match(codeAction.Title).Success)
            {
                var @namespace = AddUsingsCodeActionProviderFactory.ExtractNamespaceFromVSCSharpAddUsing(codeAction.Title);
```

seems weird to be asking one system if hte named matched, but another to grab out the namespace.  Just have this al behind a single helper.